### PR TITLE
[TEP-0091] add more no error test cases for taskrun and pipelinerun

### PR DESCRIPTION
**Pull request by Yongxuanzhang**
# Changes

This commit adds 3 more cases when the trusted resources doesn't fail the run. They are 1) no match policies and feature flag is set to ignore 2) no match policies and feature flag is set to warn 3) only warn mode policies fail.

no functional change.

/kind misc

Signed-off-by: Yongxuan Zhang yongxuanzhang@google.com

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: pkg/reconciler/pipelinerun/pipelinerun_test.go [hunk 1]
```diff
@@ -11590,30 +11635,32 @@ spec:
...
 
-	cms := []*corev1.ConfigMap{
-		{
-			ObjectMeta: metav1.ObjectMeta{Name: config.GetFeatureFlagsConfigName(), Namespace: system.Namespace()},
-			Data: map[string]string{
-				"trusted-resources-verification-no-match-policy": config.FailNoMatchPolicy,
-			},
-		},
-	}
+			cms := []*corev1.ConfigMap{
```
Comment (Yongxuanzhang) [inline pkg/reconciler/pipelinerun/pipelinerun_test.go:11638]: this code block is just move the previous test code into the for loop, and use the noMatchPolicy and verificationPolicies from the tests

**Review by lbernick — APPROVED** [review]

File: pkg/reconciler/pipelinerun/pipelinerun_test.go [hunk 2]
```diff
@@ -11580,7 +11581,51 @@ spec:
...
+	}, {
+		name:                 "only fail warn policy",
+		noMatchPolicy:        config.FailNoMatchPolicy,
+		verificationPolicies: warnPolicy,
+	},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			prs := parse.MustParseV1beta1PipelineRun(t, fmt.Sprintf(`
 metadata:
```
Comment (chuangw6) [inline pkg/reconciler/pipelinerun/pipelinerun_test.go:11629]: The `prs` seems not to change for different test cases. Can we define it outside of the for loop?

File: pkg/reconciler/taskrun/taskrun_test.go [hunk 3]
```diff
@@ -4925,7 +4926,51 @@ spec:
...
+		verificationPolicies: vps,
+	}, {
+		name:                 "only fail warn policy",
+		noMatchPolicy:        config.FailNoMatchPolicy,
+		verificationPolicies: warnPolicy,
+	},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tr := parse.MustParseV1beta1TaskRun(t, fmt.Sprintf(`
```
Comment (chuangw6) [inline pkg/reconciler/taskrun/taskrun_test.go:4973]: same for the `tr`

**Review by chuangw6 — COMMENTED** [review]
Thanks @Yongxuanzhang

File: pkg/reconciler/pipelinerun/pipelinerun_test.go (hunk 2 above)
Comment (Yongxuanzhang) [reply pkg/reconciler/pipelinerun/pipelinerun_test.go:11629]: ah I intentionally moved it inside the loop before cuz there are some issues, but now it passes. So moved it out of the loop.

File: pkg/reconciler/taskrun/taskrun_test.go (hunk 3 above)
Comment (Yongxuanzhang) [reply pkg/reconciler/taskrun/taskrun_test.go]: done

**Pull request merged**